kaldb/src/main/java/com/slack/kaldb/server/ManagerApiGrpc.java

```diff
@@ -0,0 +1,206 @@
+      ManagerApi.CreateServiceMetadataRequest request,
+      StreamObserver<Metadata.ServiceMetadata> responseObserver) {
+
+    try {
+      serviceMetadataStore.createSync(
+          new ServiceMetadata(request.getName(), request.getOwner(), 0L, Collections.emptyList()));
+      responseObserver.onNext(
+          toServiceMetadataProto(serviceMetadataStore.getNodeSync(request.getName())));
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      LOG.error("Error creating new service", e);
+      responseObserver.onError(Status.UNKNOWN.withDescription(e.getMessage()).asException());
+    }
+  }
+
+  @Override
+  public void updateServiceMetadata(
+      ManagerApi.UpdateServiceMetadataRequest request,
+      StreamObserver<Metadata.ServiceMetadata> responseObserver) {
+
+    try {
+      ServiceMetadata existingServiceMetadata = serviceMetadataStore.getNodeSync(request.getName());
+
+      ServiceMetadata updatedServiceMetadata =
+          new ServiceMetadata(
+              existingServiceMetadata.getName(),
+              request.getOwner(),
+              existingServiceMetadata.getThroughputBytes(),
+              existingServiceMetadata.getPartitionConfigs());
+      serviceMetadataStore.updateSync(updatedServiceMetadata);
+      responseObserver.onNext(toServiceMetadataProto(updatedServiceMetadata));
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      LOG.error("Error updating existing service", e);
+      responseObserver.onError(Status.UNKNOWN.withDescription(e.getMessage()).asException());
+    }
+  }
+
+  @Override
+  public void getServiceMetadata(
+      ManagerApi.GetServiceMetadataRequest request,
+      StreamObserver<Metadata.ServiceMetadata> responseObserver) {
+
+    try {
+      responseObserver.onNext(
+          toServiceMetadataProto(serviceMetadataStore.getNodeSync(request.getName())));
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      LOG.error("Error getting service", e);
+      responseObserver.onError(Status.UNKNOWN.withDescription(e.getMessage()).asException());
+    }
+  }
+
+  @Override
+  public void listServiceMetadata(
+      ManagerApi.ListServiceMetadataRequest request,
+      StreamObserver<ManagerApi.ListServiceMetadataResponse> responseObserver) {
+
+    try {
+      responseObserver.onNext(
+          ManagerApi.ListServiceMetadataResponse.newBuilder()
+              .addAllServiceMetadata(
+                  serviceMetadataStore
+                      .listSync()
+                      .stream()
+                      .map(ServiceMetadataSerializer::toServiceMetadataProto)
+                      .collect(Collectors.toList()))
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      LOG.error("Error getting services", e);
+      responseObserver.onError(Status.UNKNOWN.withDescription(e.getMessage()).asException());
+    }
+  }
+
+  @Override
+  public void updatePartitionAssignment(
```

Comment:
Please add a high level description of what this method does for various vales of the inputs.

Reply:
Done.